### PR TITLE
Add Windows AutomationId

### PR DESCRIPTION
**Pull request by nicerloop**
Expose Semantics.identifier as Windows AutomationId to ease application automation.

see https://github.com/flutter/flutter/issues/148763

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

### CODE REVIEW

**Comment by cbracken** [comment]
Hi @nicerloop; a few points.

1. It looks like you're missing a CLA. Once you've completed that step, please let us know.
2. Could you rebase this to tip-of tree please, so that it just has your changes? It looks like there are a lot of cherry-pick/etc. commits in your patch.
3. This appears to be causing some tests to segfault. Please fix the test failures.

Once these items are addressed, please let us know.

File: shell/platform/windows/flutter_window.cc [hunk 1]
```diff
@@ -806,12 +806,9 @@ LRESULT FlutterWindow::OnGetObject(UINT const message,
   }
 
   gfx::NativeViewAccessible root_view = GetNativeViewAccessible();
-  // TODO(schectman): UIA is currently disabled by default.
-  // https://github.com/flutter/flutter/issues/114547
```
Comment (loic-sharma) [inline shell/platform/windows/flutter_window.cc:810]: Unfortunately, Flutter's UIA support isn't ready to be enabled. We won't be able to land an approach that enables UIA by default.

Comment (nicerloop) [reply shell/platform/windows/flutter_window.cc:810]: With UIA disabled, automationId is not exposed. What can be done for UIA enablement ? Could it be optionally enabled when building an application ?

**Comment by yaakovschectman** [comment]
From Android triage: When the tests pass and there are no merge conflicts, we will review this.

File: shell/platform/windows/flutter_window.cc (hunk 1 above)
Comment (nicerloop) [reply shell/platform/windows/flutter_window.cc:810]: Found https://github.com/flutter/engine/pull/37754, which indicates missing parts relate to text edit carat navigation narration. Could UIA be enabled during application build, opposed to current engine build ?

File: shell/platform/common/flutter_platform_node_delegate.cc [hunk 2]
```diff
@@ -61,6 +62,11 @@ const ui::AXNodeData& FlutterPlatformNodeDelegate::GetData() const {
   return ax_node_->data();
 }
 
+std::u16string FlutterPlatformNodeDelegate::GetAuthorUniqueId() const {
+  std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t> convert;
+  return convert.from_bytes(GetData().GetStringAttribute(ax::mojom::StringAttribute::kAuthorUniqueId));
+}
```
[the remark this answers is not in the document]
Comment (nicerloop) [reply shell/platform/common/flutter_platform_node_delegate.cc:69]: Done

**Comment by nicerloop** [comment]
> Hi @nicerloop; a few points.
> 
> 1. It looks like you're missing a CLA. Once you've completed that step, please let us know.
> 2. Could you rebase this to tip-of tree please, so that it just has your changes? It looks like there are a lot of cherry-pick/etc. commits in your patch.
> 3. This appears to be causing some tests to segfault. Please fix the test failures.
> 
> Once these items are addressed, please let us know.

see https://github.com/flutter/engine/pull/53729.

**Comment by cbracken** [comment]
I'll close this since, as you note, it was replaced by https://github.com/flutter/engine/pull/53729.

**Pull request closed without merging**